File: Strings/countAndSay.py


def countAndSay(A):
    ans = '1'
    i = 1

    while i < A:
        B = str(ans)
        result = ''
        j = 0
        while j < len(B):
            count = 1
            while (j + 1 < len(B)) and B[j] == B[j+1]:
                count += 1
                j = j + 1
            result += str(count) + B[j]
            j +=1
        ans = result
        i+=1

    return result

a = 4
print(countAndSay(a))
                
            

File: Strings/countAndSay.js
function countAndSay(A) {
  ans = "1";
  i = 1;
  while (i < A) {
    B = "" + ans;
    // console.log(B)
    var result = "";
    for (var j = 0; j < B.length; j++) {
      var count = 1;
      while (B[j + 1] && B[j] === B[j + 1]) {
        count++;
        j = j + 1;
      }
      result += "" + count + B[j];
    }

    ans = result;
    i++;
  }
  return ans;
}
a = 10;
console.log(countAndSay(a));
